File: Proyecto API Jean/sportstock-api/src/main/java/com/sportstock/api/service/ProductService.java
package com.sportstock.api.service;

// ==============================
// IMPORTACIONES
// ==============================

import com.sportstock.api.model.Product;
import com.sportstock.api.repository.ProductRepository;
import org.springframework.stereotype.Service;

import java.util.List;

// ==============================
// SERVICIO DE PRODUCTOS
// Contiene la lógica de negocio
// ==============================

@Service
public class ProductService {

    // ==============================
    // REPOSITORIO
    // ==============================

    private final ProductRepository productRepository;


    // ==============================
    // INYECCIÓN DE DEPENDENCIAS
    // ==============================

    public ProductService(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }

    // ==============================
    // OBTENER TODOS LOS PRODUCTOS
    // ==============================

    public List<Product> getAllProducts() {
        return productRepository.findAll();
    }

    // ==============================
    // OBTENER PRODUCTO POR ID
    // ==============================

    public Product getProductById(Long id) {

        return productRepository.findById(id)
                .orElseThrow(() ->
                        new RuntimeException("Producto no encontrado"));
    }

    // ==============================
    // CREAR PRODUCTO
    // ==============================

    public Product createProduct(Product product) {

        // Validación básica

        if (product.getName() == null || product.getName().isBlank()) {
            throw new RuntimeException("El nombre es obligatorio");
        }

        return productRepository.save(product);
    }

    // ==============================
    // ACTUALIZAR PRODUCTO
    // ==============================

    public Product updateProduct(Long id, Product updatedProduct) {

        Product existingProduct = getProductById(id);

        existingProduct.setName(updatedProduct.getName());
        existingProduct.setCategory(updatedProduct.getCategory());
        existingProduct.setPrice(updatedProduct.getPrice());
        existingProduct.setQuantity(updatedProduct.getQuantity());
        existingProduct.setDescription(updatedProduct.getDescription());
        existingProduct.setStatus(updatedProduct.getStatus());

        return productRepository.save(existingProduct);
    }

    // ==============================
    // ELIMINAR PRODUCTO
    // ==============================

    public void deleteProduct(Long id) {

        getProductById(id);

        productRepository.deleteById(id);
    }
}

File: Proyecto tienda inventario/Prototipodeappdeinventario-main/Prototipodeappdeinventario-main/src/app/services/productService.ts
const API_URL = "http://localhost:8080/api/products";

// Obtener todos los productos
export const getProducts = async () => {
    const response = await fetch(API_URL);

    if (!response.ok) {
        throw new Error("Error al obtener productos");
    }

    return response.json();
};

// Crear producto
export const createProduct = async (product: any) => {
    const response = await fetch(API_URL, {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
        },
        body: JSON.stringify(product),
    });

    if (!response.ok) {
        throw new Error("Error al crear producto");
    }

    return response.json();
};

// Actualizar producto
export const updateProduct = async (id: number, product: any) => {
    const response = await fetch(`${API_URL}/${id}`, {
        method: "PUT",
        headers: {
            "Content-Type": "application/json",
        },
        body: JSON.stringify(product),
    });

    if (!response.ok) {
        throw new Error("Error al actualizar producto");
    }

    return response.json();
};

// Eliminar producto
export const deleteProduct = async (id: number) => {
    const response = await fetch(`${API_URL}/${id}`, {
        method: "DELETE",
    });

    if (!response.ok) {
        throw new Error("Error al eliminar producto");
    }
};